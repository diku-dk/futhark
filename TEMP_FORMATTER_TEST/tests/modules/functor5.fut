-- Open and functors must work together.
-- ==
-- input {} output {6}

module type mt = { val x: i32 }

module m1: mt = { def x = 2 }

module f(M: mt) = {
  open M
  def y = x + 2
}

module m2 = f(m1)

def main = m2.x + m2.y
