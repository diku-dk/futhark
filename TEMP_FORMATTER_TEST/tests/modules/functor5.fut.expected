45e615dd35689d9a42331d921e77bf5d
